### PR TITLE
Site Settings: Introduce a Manage Connection page with Site Ownership card

### PR DESCRIPTION
**Pull request by tyxla**
### Purpose

This PR introduces a new **Manage Connection** page under Site Settings, with the first card embedded in it - **Site Ownership**.

### Why

This PR is part of #13232, and a natural next step after we landed the necessary Redux functionality in #15599 and #15600. We will be adding the rest of the cards in subsequent PRs, and at the end, we'll add a link to this new page in the **Site Tools** section in **General** settings. Prior to that, it will be reachable from `/settings/manage-connection/$site` for Jetpack sites.

### Preview

**When current user is the connection owner**

**When another user is the connection owner**

**When the site is in development mode**

### Notes

In this initial version, there is no way to change the connection owner (contrary to the designs in #13232), this is done on purpose. We'll first need to integrate the API base for changing connection ownership, so for the first version we're going with read-only connection ownership information. *Small Moves, Ellie. Small Moves.*

### To test

* Checkout this branch, or get it going on calypso.live
* Let `:site` is one of your Jetpack sites, and you own the connection.
  * Go to `/settings/manage-connection/:site`
  * Verify you see the proper message that you're the connection owner.
* Let `:site` is one of your Jetpack sites, and somebody else owns the connection.
  * Go to `/settings/manage-connection/:site`
  * Verify you see the proper message that somebody else owns the connection.
* Let `:site` is one of your Jetpack sites that is in development mode.
  * Go to `/settings/manage-connection/:site`
  * Verify you see the expected message that your site is in dev mode and can't be connected.
* Let `:site` is one of your .com sites.
  * Go to `/settings/manage-connection/:site`
  * Verify you're redirected to the General settings section.
* Test all of the above with clean Redux state and verify they work properly.

### CODE REVIEW

File: client/my-sites/site-settings/manage-connection/index.jsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+	render() {
+		const { translate } = this.props;
+
+		return (
+			<Main className="manage-connection site-settings">
+				<DocumentHead title={ translate( 'Site Settings' ) } />
+
+				<HeaderCake onClick={ this.redirectToGeneral }>
+					{ translate( 'Manage connection' ) }
```
Comment (a8ci18n) [inline client/my-sites/site-settings/manage-connection/index.jsx]: Hi! I've found a possible matching string that has already been translated **19** times:
`translate( 'Manage Connection' )` *ES Score: 18*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/site-settings-manage-connection-page)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/my-sites/site-settings/manage-connection/site-ownership.jsx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+		const {
+			site,
+			siteIsConnected,
+			siteIsInDevMode,
+			translate,
+			userIsMaster,
+		} = this.props;
+
+		if ( siteIsConnected === false ) {
+			return translate( 'The site is not connected.' );
```
Comment (a8ci18n) [inline client/my-sites/site-settings/manage-connection/site-ownership.jsx]: Hi! I've found a possible matching string that has already been translated **21** times:
`translate( 'Your site is now connected!' )` *ES Score: 8*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/site-settings-manage-connection-page)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

Comment (tyxla) [inline client/my-sites/site-settings/manage-connection/site-ownership.jsx]: Great suggestion @a8ci18n - this could easily be a typo!

**Comment by rickybanister** [comment]
So instead of showing the `site` in this card, I intended to show the owner (if there is only one connected user) like this:

image

And if there are multiple connected admins we would show a dropdown (perhaps this is a v2 feature) so that the owner could select a new connection owner. Think of a dev setting up a site for a client and when it's finished they can make the client the owner of the connection and remove themselves without any disruption in the client's connection:

image

**Comment by tyxla** [comment]
@rickybanister sorry for the confusion, and thanks for clarifying.

The card will now show the user, and not the site.

### Updated previews

**When current user is the connection owner**


**When another user is the connection owner**


**When the site is in development mode**


Re: multiple site admins: yes, this will be a v2 feature, as we currently don't have the backend to change connection owner. 

This is ready for a new code and design review, @rickybanister @sirreal, thanks! Let's land it, so we can continue adding functionality to this screen.

**Comment by rickybanister** [comment]
Can we show the user when it is someone other than yourself? It should still be 'read-only' but if you, Marin are looking at the page and I, Rick am the owner, it would be fine to show 'Rick is the owner'. Does that make sense?

**Comment by tyxla** [comment]
@rickybanister that makes sense, but maybe for v2. Currently we don't have a way to retrieve the connection owner - we can find out whether the current user is the owner or not. I was thinking we could leave it like this for the v1?

**Comment by rickybanister** [comment]
I see. That's fine. Could you just open an issue outlining the gap in the api etc in the V2 milestone?

**Comment by tyxla** [comment]
Sure thing @rickybanister - opened #16021 for that. Thanks for the secondary review.

**Comment by tyxla** [comment]
This one can land after a second code review - perhaps @sirreal got a spare minute for that?

**Pull request merged**